doc/Building.md

```diff
@@ -0,0 +1,199 @@
+# Setting up a build environment for KOReader
+
+These instructions are intended to build the emulator in Linux and MacOS. Windows users are suggested to develop in a [Linux VM](https://www.howtogeek.com/howto/11287/how-to-run-ubuntu-in-windows-7-with-vmware-player/) or using the [Windows Subsystem for Linux](https://en.wikipedia.org/wiki/Windows_Subsystem_for_Linux).
+
+If you only want to work with Lua frontend stuff, you can grab the AppImage and
+run it with `--appimage-extract`.
+
+You can skip most of the following instructions if desired, and use our premade Docker image instead. In that case the only requirements are Git and Docker. See [the virtual development environment README](https://github.com/koreader/virdevenv) for more information.
+
+## Prerequisites
+
+To get and compile the source you must have `patch`, `wget`, `unzip`, `git`,
+`cmake` and `luarocks` installed, as well as a version of `autoconf`
+greater than 2.64. You also need `nasm` and of course a compiler like `gcc`
+or `clang`. If you want to cross-compile for other architectures, you need a proper
+cross-compile toolchain. Your GCC should be at least version 4.8.
+
+### Debian/Ubuntu and derivates
+
+Install the prerequisites using dpkg:
```

Comment:
dpkg? Perhaps apt uses `dpkg -i` behind the scenes but for an aspiring hacker that's just adding unnecessary confusion. Here it says dpkg, then it says apt-get? Where does dpkg come in?
```suggestion
Install the prerequisites using APT:
```